calico/openstack/test/test_plugin.py

```diff
@@ -0,0 +1,458 @@
+        def make_recv(name, socket):
+
+            def recv(*args):
+                print "Socket %s recv_%s..." % (socket, name)
+
+                #*************************************************************#
+                #* Block until there's something to receive, and then get    *#
+                #* that.                                                     *#
+                #*************************************************************#
+                msg = socket.rcv_queue.get(True)
+
+                #*************************************************************#
+                #* Return that.                                              *#
+                #*************************************************************#
+                return msg
+
+            return recv
+
+        #*********************************************************************#
+        #* When socket calls poll, block on the socket's receive queue.      *#
+        #*********************************************************************#
+        def make_poll(socket):
+
+            def poll(ms):
+                print "Socket %s poll for %sms..." % (socket, ms)
+
+                #*************************************************************#
+                #* Block until there's something to receive, and then get    *#
+                #* that.                                                     *#
+                #*************************************************************#
+                msg = socket.rcv_queue.get(True, ms / 1000)
+
+                #*************************************************************#
+                #* Get the message back on the queue, for a following        *#
+                #* receive call.                                             *#
+                #*************************************************************#
+                socket.rcv_queue.put_nowait(msg)
+                self.real_eventlet_sleep(REAL_EVENTLET_SLEEP_TIME)
+
+                #*************************************************************#
+                #* Return nothing.                                           *#
+                #*************************************************************#
+                return None
+
+            return poll
+
+        #*********************************************************************#
+        #* Print logs to stdout.                                             *#
+        #*********************************************************************#
+        def log_info(msg):
+            print ">>>>>>>INFO %s" % msg
+            return None
+        def log_debug(msg):
+            print ">>>>>>DEBUG %s" % msg
+            return None
+        def log_warn(msg):
+            print ">>>>>>>WARN %s" % msg
+            return None
+        def log_exception(msg):
+            print ">>EXCEPTION %s" % msg
+            traceback.print_exc()
+            return None
```

Comment:
None of this should be necessary: `nosetests` will intercept logging and display it for failed tests.